### DISCUSSION TITLE
Problems encountered when gmsh generates grids

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#672] [Q&A] [answered]
I can successfully generate the grid I need using the following script,

![image](https://user-images.githubusercontent.com/104721358/191024323-3e8cbd85-e600-41bf-9697-352316545240.png)

![image](https://user-images.githubusercontent.com/104721358/191024141-bc834c2c-d53d-4cc5-b05c-9abbef4f6e91.png)

But when I use the gmsh GUI to open the same geo file to generate the .msh file through file --> export and then use the following command to convert the .msh file exported by the GUI to a puml.h5 file, it has no response.
![QQ截图20220919210038](https://user-images.githubusercontent.com/104721358/191023114-055c8318-e541-4dcc-90b4-b636dbf13179.png)

![image](https://user-images.githubusercontent.com/104721358/191025877-63a9e945-12db-4192-9f59-c23d3d39ca45.png)

![image](https://user-images.githubusercontent.com/104721358/191026118-445b0f53-1908-4c23-b6ad-2d88b2082245.png)

![image](https://user-images.githubusercontent.com/104721358/191025433-ca032385-3001-4298-beee-4d5bcaca6c0a.png)

Then I used a very simple geo model to test it, and it was able to generate meshes, but when I converted it to puml.h5, I encountered the following problem. Why does this problem occur?
![image](https://user-images.githubusercontent.com/104721358/191028417-8c1b4cd5-8a67-4b76-815c-79711f0da8e9.png)

![image](https://user-images.githubusercontent.com/104721358/191028229-c777adab-8c6e-4d9d-a75a-29fc049499af.png)

![image](https://user-images.githubusercontent.com/104721358/191028933-2481c28d-5345-412f-879a-d251b3d73285.png)

**Comment by Thomas-Ulrich** [comment]
Well your very simple example only has surface elements, while pumgen expect volume elements.
In gmsh GUI, you also have to click on Mesh->Define->3D.
Did you do that?

**Comment by smallriver666** [comment]
I try it, but it doesn't work,I found a difference between the GUI and the script exported meshes，



the GUI


![image](https://user-images.githubusercontent.com/104721358/195516571-b36c0750-2640-4b42-ba8a-54cffc7ca28b.png)



the script


![image](https://user-images.githubusercontent.com/104721358/195517174-e651002a-95d2-4091-ad8b-9987bf128cc0.png)

By the way, I would like to ask if the parameter `3` in `gmsh -3` is to generate a 3D mesh?
I first express my sincere thanks

**Comment by smallriver666** [comment, marked as answer]
The problem has solved, may just because the problem of my computer, I use another computer, it works perfectly.
